utils/build-script-impl

```diff
@@ -827,6 +827,12 @@ function set_build_options_for_host() {
     # and it must be the same for both LLVM and Swift
 
     if [[ "${SWIFT_HOST_TRIPLE}" ]] ; then
+        # Adding this until we can configure these corelibs repos with a CMake
+        # toolchain file
+        SWIFT_TARGET_CMAKE_OPTIONS+=(
+            -DCMAKE_Swift_COMPILER_TARGET:STRING="${SWIFT_HOST_TRIPLE}"
```

Comment:
Recent corelibs changes to use `CMAKE_Swift_COMPILER_TARGET` broke cross-compiling with `build-script`. This wasn't noticed on the CI because the only one that cross-compiles the whole toolchain, the macOS CI, doesn't build corelibs like libdispatch or Foundation. I've been working around this for months on my Android CI, finagolfin/swift-android-sdk@774f3b612, but this fixes it so I can remove those.